benchmark/config.h

```diff
@@ -0,0 +1,39 @@
+#ifndef BENCHMARK_CONFIG_H_
+#define BENCHMARK_CONFIG_H_
+
+#include <stdint.h>
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "benchmark/vendor.h"
+
+#include "tachyon/base/flag/flag_parser.h"
+
+namespace tachyon::benchmark {
+
+class Config {
+ public:
+  struct Options {
+    bool include_vendors = false;
+  };
+
+  Config() = default;
+  Config(const Config& other) = delete;
+  Config& operator=(const Config& other) = delete;
+
+  const std::vector<benchmark::Vendor>& vendors() const { return vendors_; }
+  bool check_results() const { return check_results_; }
+
+  bool Parse(int argc, char** argv, const Options& options,
+             base::FlagParser* parser = nullptr);
```

Comment:
I wanted `XXConfig::Parse()` to call `Config::Parse()` after adding some custom flags to reduce code duplicates. But one problem with that is we cannot overwrite help messages. I wasn't sure how to handle it.
1. modify `FlagParser` so it can handle it
2. store help messages per flag in `Config` and set it from `XXConfig`.

We need this mainly for printing supported vendors.

Reply:
I think it's better to make a new class like `FlagParserBase` under `benchmark` directory and make it have default flags, and then enable it to be configured with a child class.

```c++
namespace tachyon::benchmark {

// in benchmark/flag_parser_base.h
class FlagPaserBase : public base::FlagParser {
 public:
  struct Options {
    bool include_vendors;
  };

  FlagPaserBase(const Options& options) {
    AddFlag<base::BoolFlag>(&check_results_)
      .set_long_name("--check_results")
      .set_help("Check results across different vendors.");
    if (options.include_vendors) {
      AddFlag<base::Flag<std::vector<Vendor>>>(&vendors_)
        .set_long_name("--vendor")
        .set_help("Vendors to be benchmarked with.");
    }
  }

 protected:
  bool check_results_;
  std::vector<Vendor> vendors_;
};

// in benchmark/xx/flag_parser.h
class XXXFlagParser : public FlagParserBase {
 public:
  struct Options: FlagParserBaseOptions {
     // ...
  };

  XXXFlagParser(const Options& options) : FlagParserBase(options) {
    // .. 
  }
}
```